dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -11,20 +11,41 @@
        for (presumably) faster overall incremental processing.
 
        Low-level specifics:
+       If an invalid option is specified, then we will raise an
+       excpetion with corresponding message.
+
        Languages other than SQL (like Python) will use a temporary table.
        With the default strategy of merge, the user may choose between a temporary
        table and view (defaulting to view).
+
        The append strategy can use a view because it will run a single INSERT statement.
-       When the unique_key is none,
-       then we can use a view because it will run a single INSERT statement.
+
+       When unique_key is none, the delete+insert strategy can use a view beacuse a
+       single INSERT statement is run with no DELETES as part of the statement.
        Otherwise, play it safe by using a temporary table.
   #} */
 
+  {% if language == "python" and tmp_relation_type is not none %}
+    {% do exceptions.raise_compiler_error(
+      "Python models currently only support 'table' for tmp_relation_type but "
+       ~ tmp_relation_type ~ " was specified."
+    ) %}
+  {% endif %}
+
+  {% if strategy == "delete+insert" and tmp_relation_type != "table" and unique_key is not none %}
```

Comment:
The CI output [here](https://github.com/dbt-labs/dbt-snowflake/actions/runs/3952937214/jobs/6768632525) gives us some hints of what we'd need to change:
```
21:10:02  Completed with 1 error and 0 warnings:
21:10:02  
21:10:02  Compilation Error in model model (models-unquoted/model.sql)
21:10:02    In order to maintain consistent results when `unique_key` is not none,
21:10:02          the `delete+insert` strategy only supports `table` for `tmp_relation_type` but None was specified.
```

Although a bit hard to read, using the [Jinja `default` filter](https://jinja.palletsprojects.com/en/3.0.x/templates/#jinja-filters.default) might work:
```suggestion
  {% if strategy == "delete+insert" and (tmp_relation_type|default("table") != "table" and unique_key is not none %}
```

A different way to write the same thing:
```suggestion
  {% if strategy == "delete+insert" and tmp_relation_type is not none and tmp_relation_type != "table" and unique_key is not none %}
```

Reply:
like minds, had the second implementation on my list of things to try, currently trying moving the exceptions after our actual choice logic that way it's testing after we choose rather than before just want to make sure the main condtional logic works the way we think as well.

Reply:
some quick testing makes it seem like `None` counts as a valid value and so wouldn't swap out to the default in the filter method

testing second method, now didn't like how the moving of the exceptions acted.